frontend/packages/topology/src/components/workload/useResourceWatcher.ts

```diff
@@ -0,0 +1,17 @@
+import * as React from 'react';
+import { useK8sWatchResource } from '@console/internal/components/utils/k8s-watch-hook';
+import { K8sResourceCommon } from '@console/internal/module/k8s';
+import { getResourceFromWatchResults } from './utils';
+
+export const useResourceWatcher = <R extends K8sResourceCommon>(resourceData: R) => {
+  const watchKind = resourceData.kind;
+  const { namespace, uid } = resourceData.metadata;
+  const watchedResourceObj = React.useMemo(() => ({ isList: true, kind: watchKind, namespace }), [
+    watchKind,
+    namespace,
+  ]);
+  const [watchedResources, loaded, loadError] = useK8sWatchResource<R[]>(watchedResourceObj);
+  const resource: R =
+    watchedResources.length > 0 ? getResourceFromWatchResults<R>(uid, watchedResources) : null;
```

Comment:
Based on my previous comment, `getResourceFromWatchResults` function can be removed.